src/gui/transferlistwidget.cpp

```diff
@@ -363,19 +363,33 @@ void TransferListWidget::permDeleteSelectedTorrents()
     deleteSelectedTorrents(true);
 }
 
-void TransferListWidget::deleteSelectedTorrents(bool deleteLocalFiles)
+void TransferListWidget::deleteSelectedTorrents(const bool deleteLocalFiles)
 {
     if (m_mainWindow->currentTabWidget() != this) return;
 
     const QVector<BitTorrent::TorrentHandle *> torrents = getSelectedTorrents();
     if (torrents.empty()) return;
 
-    if (Preferences::instance()->confirmTorrentDeletion()
-        && !DeletionConfirmationDialog::askForDeletionConfirmation(this, deleteLocalFiles, torrents.size(), torrents[0]->name()))
-        return;
-    const DeleteOption deleteOption = deleteLocalFiles ? TorrentAndFiles : Torrent;
-    for (const BitTorrent::TorrentHandle *torrent : torrents)
-        BitTorrent::Session::instance()->deleteTorrent(torrent->hash(), deleteOption);
+    const auto removeTorrents = [torrents](const bool isDeleteFileSelected)
```

Comment:
I wouldn't use lambda in this case. Why not make it a private method?

Reply:
OK, done.